Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+import Checkout
+
+protocol ExpiryDateViewDelegate: AnyObject {
+    func update(expiryDate: ExpiryDate)
+}
+
+final class ExpiryDateView: UIView {
+    weak var delegate: ExpiryDateViewDelegate?
+
+    private(set) var style: CellTextFieldStyle?
+
+    private lazy var cardValidator: CardValidator = {
+        CardValidator(environment: .production)
+    }()
+
+    private lazy var expiryDateView: InputView = {
+        let view = InputView().disabledAutoresizingIntoConstraints()
+        view.delegate = self
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        // setup expiry DateView
+        addSubview(expiryDateView)
+        expiryDateView.setupConstraintEqualTo(view: self)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func update(style: CellTextFieldStyle?) {
+        self.style = style
+        expiryDateView.update(style: style)
+    }
+
+    private func updateErrorView(isHidden: Bool, text: String?){
+        style?.error?.isHidden = isHidden
+        style?.textfield.text = text ?? ""
+        expiryDateView.update(style: style)
+    }
+
+    @discardableResult
+    func isValidExpiryDate(text: String?) -> Bool {
+        let subString = text?.split(separator: "/")
+        guard let month = subString?.first,
```

Comment:
I would add another test case for input of `01/01/01/01`. Then to pass that test we'd need to check the count of `subString`.